### PR TITLE
Document how to prepare a release

### PR DESCRIPTION
**Pull request by marcofucci**
### Description of change

This adds a bit of documentation on how to prepare a release.

### Checklist

* [ ] Has a new newsfragment been created? Check [changelog/README.rst](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.rst) for instructions
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: docs/How to prepare a release.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+Add and push the changes:
+
+```
+git add .
+git commit -m 'Add changelog for <version>'
+git push origin changelog-<version>
+```
+
+In GitHub open a PR to merge `changelog-<version>` into `develop` and assign at least two developers for review.
```
Comment (elcct) [inline docs/How to prepare a release.md]: `In GitHub, ` - missing comma

File: docs/How to prepare a release.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# How to prepare a release
+
+Decide the version number for the release using the format `<major>.<minor>.<patch>`.
```
Comment (reupen) [inline docs/How to prepare a release.md]: Would add a subheading here (e.g. 'Decide on the new version number') as I had to read these two sentences twice to realise that they're connected.

File: docs/How to prepare a release.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+After the PR has been reviewed, merge it into `master` and delete the merged branch.
+The release will be automatically deployed to staging via Jenkins.
+Releasing to production has to happen manually after it's been tested on staging and approved by the Live Services team.
+
+## Tag
+
+From your local environment checkout and pull `master`, create and push a tag for the release:
+
+```
```
Comment (reupen) [inline docs/How to prepare a release.md]: Creating a release on GitHub will create a tag as well so I'm not sure there's much point creating the tag locally first and pushing it.

But if we do create it locally, it should be an annotated tag (so it has a date and author etc.)

Comment (marcofucci) [reply docs/How to prepare a release.md]: If you create it locally and you draft a release in GitHub, the tag appears in the dropdown. I'll get rid of this section altogether maybe to make it simpler.

Comment (reupen) [reply docs/How to prepare a release.md]: I agree in that I would make the release section a bit more explicit about what to put in each box and it should mention how the tag gets created.

I'm not sure why you'd want to create the tag locally first either when it's something that is frequently done incorrectly.

Comment (marcofucci) [reply docs/How to prepare a release.md]: Make sense, thanks both.

**Pull request merged**